torch/_inductor/select_algorithm.py

```diff
@@ -1752,16 +1754,35 @@ def no_op(*args, **kwargs):
 
             def precompile_with_captured_stdout(choice):
                 with restore_stdout_stderr(initial_stdout, initial_stderr):
-                    start_time = time.time()
                     choice.precompile()
-                    return time.time() - start_time
+
+            def on_complete(future):
+                assert future in start_times
+                elapsed_times[future] = time.time() - start_times[future]
 
             executor = ThreadPoolExecutor(max_workers=num_workers)
+            async_compile = torch._inductor.async_compile.AsyncCompile()
+
+            futures: Dict[concurrent.futures.Future[Any], ChoiceCaller] = {}
+            start_times: Dict[concurrent.futures.Future[Any], float] = {}
+            elapsed_times: Dict[concurrent.futures.Future[Any], float] = {}
 
-            futures = {}
             for c in choices:
                 if hasattr(c, "precompile"):
-                    future = executor.submit(precompile_with_captured_stdout, c)
+                    triton_cuda_choice = isinstance(
+                        c, TritonTemplateCaller
+                    ) and isinstance(c.bmreq, TritonGPUBenchmarkRequest)
+                    if triton_cuda_choice and async_compile._use_process_pool():
```

Comment:
nit: wanna remove the leading underscore on `_use_process_pool()` since it's no longer a private method? Alternatively, we could make async_compile.triton just do the right thing? Or altnernatively, create a new method like `async_compile.triton_suprocess` that `async_compile.triton` calls? It's just a little weird to me that `async_compile.triton()` has a code branch we'll never call because we check `use_process_pool` before calling int.

Reply:
This is part of compile-time autotuning, correct? If so, this is probably going to make the scuba logging in https://github.com/pytorch/pytorch/pull/143104 incorrect. Previously, we could just time each operation and add accumulate into a value that we log to scuba at the end of compilation. But if part of the autotuning can happen in parallel, it will over count.

Reply:
> async_compile.triton() has a code branch we'll never execute because we check use_process_pool before calling

This is true in this current path but not generally. I don't know the other callees..

Going to remove the leading underscore for now.

Reply:
> If so, this is probably going to make the scuba logging in https://github.com/pytorch/pytorch/pull/143104 incorrect
> But if part of the autotuning can happen in parallel, it will over count

This was already the case. we're now using process pool and we were previously using threadpool